source/code/plugins/out_oms_diag.rb

```diff
@@ -0,0 +1,112 @@
+module Fluent
+
+  class OutputOMSDiag < BufferedOutput
+
+    Plugin.register_output('out_oms_diag', self)
+
+    def initialize
+      super
+      require 'net/http'
+      require 'net/https'
+      require 'uri'
+      require_relative 'omslog'
+      require_relative 'oms_configuration'
+      require_relative 'oms_common'
+      require_relative 'oms_diag'
+    end
+
+    config_param :omsadmin_conf_path, :string, :default => '/etc/opt/microsoft/omsagent/conf/omsadmin.conf'
+    config_param :cert_path, :string, :default => '/etc/opt/microsoft/omsagent/certs/oms.crt'
+    config_param :key_path, :string, :default => '/etc/opt/microsoft/omsagent/certs/oms.key'
+    config_param :proxy_conf_path, :string, :default => '/etc/opt/microsoft/omsagent/proxy.conf'
+    # TODO: To make compress default to true once supported
+    config_param :compress, :bool, :default => false
+
+    def configure(conf)
+      super
+    end
+
+    def start
+      super
+      @proxy_config = OMS::Configuration.get_proxy_config(@proxy_conf_path)
+    end
+
+    def shutdown
+      super
+    end
+
+    def handle_record(ipname, record)
+      @log.trace "Handling diag records for ipname : #{ipname}"
```

Comment:
For logged messages, I would recommend spelling out diagnostic, for added clarity. Same for lines 51, 55, 56, 63, 70, etc.